### PR TITLE
Tooltips for collapsed navigation

### PR DESCRIPTION
**Pull request by pxlrbt**
Add tooltips to navigation items when it is collapsed.

https://user-images.githubusercontent.com/22632550/159532733-9b5d496e-b13a-4a5f-93e3-d893dd4f3c7f.mov

Should we have light tooltips on dark theme and dark tooltips on light theme for better contrast or the other way around?

### CODE REVIEW

File: packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php [hunk 1]
```diff
@@ -11,6 +11,21 @@
...
         {!! $shouldOpenUrlInNewTab ? 'target="_blank"' : '' !!}
         x-on:click="window.matchMedia(`(max-width: 1024px)`).matches && $store.sidebar.close()"
+        x-data="{tooltip: {}}"
+        x-init="
+            Alpine.effect(() => {
+                if (Alpine.store('sidebar').isOpen) {
+                    tooltip = false
+                } else {
+                    tooltip = {
+                        content: '{{ $slot }}',
```
Comment (ryangjchandler) [inline packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php:21]:
```suggestion
                        content: @js($slot),
```

**Review by ryangjchandler — CHANGES_REQUESTED** [review]

File: packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php (hunk 1 above)
Comment (ryangjchandler) [reply packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php:21]: You should be able to do something like `@js((string) $slot)`

Comment (pxlrbt) [reply packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php:21]: Slots are `HtmlString`s. I could do `@js($slot->toHtml())` if that's better.

Comment (ryangjchandler) [reply packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php:21]: Yeah, either or. I'm pretty sure `HtmlString` has a `__toString()` method as well, but if not yeah, `toHtml()`.

Comment (pxlrbt) [reply packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php:21]: Casting or `->toHtml()` what do you prefer? I think I'd go with the more explicit function.

Comment (ryangjchandler) [reply packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php:21]: Yeah, `toHtml()` then.

File: packages/admin/resources/js/app.js [hunk 2]
```diff
@@ -39,6 +39,18 @@ Alpine.store('sidebar', {
...
+    window.matchMedia("(prefers-color-scheme: dark)").matches ? 'dark' : 'light'
+);
+
+window
+    .matchMedia("(prefers-color-scheme: dark)")
+    .addEventListener("change", (e) => {
+        Alpine.store('theme', e.matches ? 'dark' : 'light')
+    });
+
+
```
Comment (danharrin) [inline packages/admin/resources/js/app.js:53]:
```suggestion
```

File: packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php [hunk 3]
```diff
@@ -11,6 +11,21 @@
...
         x-on:click="window.matchMedia(`(max-width: 1024px)`).matches && $store.sidebar.close()"
+        x-data="{tooltip: {}}"
+        x-init="
+            Alpine.effect(() => {
+                if (Alpine.store('sidebar').isOpen) {
+                    tooltip = false
+                } else {
+                    tooltip = {
+                        content: @js($slot->toHtml()),
+                        theme:  Alpine.store('theme') === 'light' ? 'dark' : 'light',
```
Comment (danharrin) [inline packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php:22]:
```suggestion
                        theme: Alpine.store('theme') === 'light' ? 'dark' : 'light',
```

File: packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php [hunk 4]
```diff
@@ -11,6 +11,21 @@
...
+                        placement: document.dir === 'rtl' ? 'left' : 'right'
```
Comment (danharrin) [inline packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php:23]:
```suggestion
                        placement: document.dir === 'rtl' ? 'left' : 'right',
```

File: packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php [hunk 5]
```diff
@@ -11,6 +11,21 @@
         href="{{ $url }}"
         {!! $shouldOpenUrlInNewTab ? 'target="_blank"' : '' !!}
         x-on:click="window.matchMedia(`(max-width: 1024px)`).matches && $store.sidebar.close()"
+        x-data="{tooltip: {}}"
```
Comment (danharrin) [inline packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php:14]:
```suggestion
        x-data="{ tooltip: {} }"
```

File: packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php [hunk 6]
```diff
@@ -11,6 +11,21 @@
...
+                if (Alpine.store('sidebar').isOpen) {
+                    tooltip = false
+                } else {
+                    tooltip = {
+                        content: @js($slot->toHtml()),
+                        theme:  Alpine.store('theme') === 'light' ? 'dark' : 'light',
+                        placement: document.dir === 'rtl' ? 'left' : 'right'
+                    }
+                }
+            });
```
Comment (danharrin) [inline packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php:26]:
```suggestion
            })
```

File: packages/admin/resources/js/app.js [hunk 7]
```diff
@@ -39,6 +39,18 @@ Alpine.store('sidebar', {
...
+Alpine.store(
+    'theme',
+    window.matchMedia("(prefers-color-scheme: dark)").matches ? 'dark' : 'light'
+);
+
+window
+    .matchMedia("(prefers-color-scheme: dark)")
+    .addEventListener("change", (e) => {
+        Alpine.store('theme', e.matches ? 'dark' : 'light')
+    });
```
Comment (danharrin) [inline packages/admin/resources/js/app.js:51]:
```suggestion
    })
```

File: packages/admin/resources/js/app.js [hunk 8]
```diff
@@ -39,6 +39,18 @@ Alpine.store('sidebar', {
...
 
+Alpine.store(
+    'theme',
+    window.matchMedia("(prefers-color-scheme: dark)").matches ? 'dark' : 'light'
+);
+
+window
+    .matchMedia("(prefers-color-scheme: dark)")
+    .addEventListener("change", (e) => {
+        Alpine.store('theme', e.matches ? 'dark' : 'light')
```
Comment (danharrin) [inline packages/admin/resources/js/app.js:50]:
```suggestion
        Alpine.store('theme', event.matches ? 'dark' : 'light')
```

File: packages/admin/resources/js/app.js [hunk 9]
```diff
@@ -39,6 +39,18 @@ Alpine.store('sidebar', {
     },
 })
 
+Alpine.store(
+    'theme',
+    window.matchMedia("(prefers-color-scheme: dark)").matches ? 'dark' : 'light'
+);
+
+window
+    .matchMedia("(prefers-color-scheme: dark)")
```
Comment (danharrin) [inline packages/admin/resources/js/app.js:48]:
```suggestion
    .matchMedia('(prefers-color-scheme: dark)')
```

File: packages/admin/resources/js/app.js [hunk 10]
```diff
@@ -39,6 +39,18 @@ Alpine.store('sidebar', {
     },
 })
 
+Alpine.store(
+    'theme',
+    window.matchMedia("(prefers-color-scheme: dark)").matches ? 'dark' : 'light'
```
Comment (danharrin) [inline packages/admin/resources/js/app.js:44]:
```suggestion
    window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light',
```

File: packages/admin/resources/css/app.css [hunk 11]
```diff
@@ -1,5 +1,6 @@
 @import '../../../forms/dist/module.esm.css';
```
Comment (danharrin) [inline packages/admin/resources/css/app.css:1]:
```suggestion
@import '../../../forms/dist/module.esm.css';

```

File: packages/admin/resources/js/app.js [hunk 12]
```diff
@@ -39,6 +39,17 @@ Alpine.store('sidebar', {
     },
 })
 
+Alpine.store(
+    'theme',
+    window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light',
+);
```
Comment (danharrin) [inline packages/admin/resources/js/app.js:45]:
```suggestion
)
```

**Review by ryangjchandler — APPROVED** [review]

**Comment by danharrin** [comment]
Thanks! What are we gonna do about toggling dark mode for other tooltips?

**Comment by ryangjchandler** [comment]
@danharrin I believe somebody is already working on that.

**Comment by danharrin** [comment]
By the way, when I just tested this it all of the tooltips were white, regardless of theme

**Comment by ryangjchandler** [comment]
@danharrin Seems to be working fine for me..

**Comment by SebastiaanKloos** [comment]
@pxlrbt @danharrin this does not work with custom filament themes :(

```php
Filament::serving(function (): void {
    Filament::registerTheme(mix('css/app.css'));
});
```

**Pull request merged**